packages/edit-post/src/hooks/components/media-upload/index.js

```diff
@@ -80,6 +108,7 @@ class MediaUpload extends Component {
 		type: deprecatedType,
 		multiple = false,
 		gallery = false,
+		featuredImage = false,
```

Comment:
This, along with the `gallery` property, seems odd to me, since they are mutually exclusive. We have other flow-specific frame states that might need to be supported both here and in other places in core (e.g., media widgets). I wonder if we could create those specific frames pass a `context` or `type` property to `MediaUpload` rather than several of these. See #1979.

Reply:
yeah, I have another one that I am passing arguments similar for the `classic-gallery` block: https://github.com/WordPress/gutenberg/compare/master...antpb:add/classic-gallery#diff-781b45da3dbed5d2670d2c98e473fb83

It's hard to do an audit of all the frame states expected in core because they are kinda scattered around and in some cases conditional. I'm noticing the 5.1 label on #1979. Do you think we could go forward with this flow for 5.0 and improve later? I think the add media tabs issue may also change the nature of this.

Reply:
It is probably much easier to see all of the states in the development repo, rather than the core build repo. See: https://github.com/WordPress/wordpress-develop/tree/master/src/js/media